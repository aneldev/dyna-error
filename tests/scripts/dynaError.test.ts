import {
  dynaError,
  IDynaError,
} from "../../src";

describe('dynaError', () => {
  describe('with error config', () => {
    it('Minimal use, just with a message', () => {
      try {
        throw dynaError({message: 'Something is invalid'});
      }
      catch (e) {
        const error: IDynaError = e;
        expect(error).toMatchSnapshot();
        expect((error.stack || '').length).toBeGreaterThan(0);
        expect(error.isDynaError).toBe(true);
        expect(error.date).not.toBe(undefined);
        expect(error.date.valueOf()).toBeGreaterThan(0);
        expect(error.message).toBe('Something is invalid');
        expect(error.userMessage).toBe(undefined);
        expect(error.code).toBe(undefined);
        expect(error.status).toBe(undefined);
        expect(error.data).toBe(undefined);
        expect(error.parentError).toBe(undefined);
        expect(error.validationErrors).toBe(undefined);
        expect(error.canRetry).toBe(undefined);
      }
    });
    it('Use with all properties', () => {
      try {
        throw dynaError({
          message: 'Something is invalid',
          userMessage: 'Please retry',
          code: 330010,
          status: 500,
          canRetry: false,
          parentError: {message: 'Parent error'},
          validationErrors: {name: 'Is required'},
          data: {userId: 230130042},
        });
      }
      catch (e) {
        const error: IDynaError = e;
        expect(error).toMatchSnapshot();
        expect((error.stack || '').length).toBeGreaterThan(0);
        expect(error.isDynaError).toBe(true);
        expect(error.date).not.toBe(undefined);
        expect(error.date.valueOf()).toBeGreaterThan(0);
        expect(error.parentError.message).toBe('Parent error');
        expect(error.validationErrors?.name).toBe('Is required');
        expect(error.message).toBe('330010 Something is invalid');
        expect(error.userMessage).toBe('Please retry');
        expect(error.code).toBe(330010);
        expect(error.status).toBe(500);
        expect(error.canRetry).toBe(false);
        expect(error.data.userId).toBe(230130042);
      }
    });
  });
  describe('with string argument', () => {
    test('Error with string arg', () => {
      try {
        throw dynaError('Something is invalid');
      }
      catch (e) {
        const error: IDynaError = e;
        expect(error).toMatchSnapshot();
        expect((error.stack || '').length).toBeGreaterThan(0);
        expect(error.isDynaError).toBe(true);
        expect(error.date).not.toBe(undefined);
        expect(error.date.valueOf()).toBeGreaterThan(0);
        expect(error.message).toBe('Something is invalid');
        expect(error.userMessage).toBe(undefined);
        expect(error.code).toBe(undefined);
        expect(error.status).toBe(undefined);
        expect(error.data).toBe(undefined);
        expect(error.canRetry).toBe(undefined);
      }
    });
  });
});
