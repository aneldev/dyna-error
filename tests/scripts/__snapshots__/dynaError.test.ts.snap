// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynaError From native Error 1`] = `
Object {
  "isDynaError": true,
  "message": "Something went wrong",
  "name": "Error",
}
`;

exports[`dynaError with error config Minimal use, just with a message 1`] = `
Object {
  "isDynaError": true,
  "message": "Something is invalid",
  "name": "Error",
}
`;

exports[`dynaError with error config Use with all properties 1`] = `
Object {
  "canRetry": false,
  "code": 330010,
  "data": Object {
    "userId": 230130042,
  },
  "isDynaError": true,
  "message": "330010: Something is invalid",
  "name": "Error",
  "parentError": Object {
    "message": "Parent error",
  },
  "status": 500,
  "userData": Object {
    "level": "basic",
  },
  "userMessage": "Please retry",
  "validationErrors": Object {
    "name": "Is required",
  },
}
`;

exports[`dynaError with string argument Error with string arg 1`] = `
Object {
  "isDynaError": true,
  "message": "Something is invalid",
  "name": "Error",
}
`;
